dotnet/src/IntegrationTests/Connectors/OpenAI/OpenAIChatCompletion_RequiredFunctionChoiceBehaviorTests.cs

```diff
@@ -0,0 +1,399 @@
+    {
+        // Arrange
+        var plugin = this._kernel.CreatePluginFromType<DateTimeUtils>(); // Creating plugin without importing it to the kernel.
+
+        var invokedFunctions = new List<string>();
+
+        this._autoFunctionInvocationFilter.RegisterFunctionInvocationHandler(async (context, next) =>
+        {
+            invokedFunctions.Add(context.Function.Name);
+            await next(context);
+        });
+
+        var functionsForManualInvocation = new List<string>();
+
+        var settings = new OpenAIPromptExecutionSettings { FunctionChoiceBehavior = FunctionChoiceBehavior.Required([plugin.ElementAt(0)], autoInvoke: false) };
+
+        var chatHistory = new ChatHistory();
+        chatHistory.AddUserMessage("How many days until Christmas?");
+
+        // Act
+        await foreach (var content in this._chatCompletionService.GetStreamingChatMessageContentsAsync(chatHistory, settings, this._kernel))
+        {
+            if (content is OpenAIStreamingChatMessageContent openAIContent && openAIContent.ToolCallUpdates is { Count: > 0 } && !string.IsNullOrEmpty(openAIContent.ToolCallUpdates[0].FunctionName))
+            {
+                functionsForManualInvocation.Add(openAIContent.ToolCallUpdates[0].FunctionName);
+            }
+        }
+
+        // Assert
+        Assert.Single(functionsForManualInvocation);
+        Assert.Contains("DateTimeUtils-GetCurrentDate", functionsForManualInvocation);
+
+        Assert.Empty(invokedFunctions);
+    }
+
+    private Kernel InitializeKernel()
+    {
+        var openAIConfiguration = this._configuration.GetSection("OpenAI").Get<OpenAIConfiguration>();
+        Assert.NotNull(openAIConfiguration);
+        Assert.NotNull(openAIConfiguration.ChatModelId!);
+        Assert.NotNull(openAIConfiguration.ApiKey);
+
+        var kernelBuilder = base.CreateKernelBuilder();
+
+        kernelBuilder.AddOpenAIChatCompletion(
+            modelId: openAIConfiguration.ChatModelId,
+            apiKey: openAIConfiguration.ApiKey);
+
+        return kernelBuilder.Build();
+    }
+
+    private readonly IConfigurationRoot _configuration = new ConfigurationBuilder()
+        .AddJsonFile(path: "testsettings.json", optional: false, reloadOnChange: true)
+        .AddJsonFile(path: "testsettings.development.json", optional: true, reloadOnChange: true)
+        .AddEnvironmentVariables()
+        .AddUserSecrets<OpenAIChatCompletionTests>()
+        .Build();
+
+    /// <summary>
+    /// A plugin that returns the current time.
+    /// </summary>
+#pragma warning disable CA1812 // Avoid uninstantiated internal classes
+    private sealed class DateTimeUtils
+#pragma warning restore CA1812 // Avoid uninstantiated internal classes
+    {
+        [KernelFunction]
+        [Description("Retrieves the current date.")]
+        public string GetCurrentDate() => DateTime.UtcNow.ToString("d", CultureInfo.InvariantCulture);
+    }
+
+    #region private
```

Comment:
Move this `region` above `DateTimeUtils` class.

Reply:
Done, even though I doubt that the region provides any value.